src/manifolds/Euclidean.jl

```diff
@@ -275,6 +290,18 @@ function local_metric(::MetricManifold{𝔽,<:Manifold,EuclideanMetric}, p) wher
     return Diagonal(ones(SVector{size(p, 1),eltype(p)}))
 end
 
+function inverse_retract(M::Euclidean{Tuple{}}, x::T, y::T) where {T<:Number}
+    return inverse_retract(M, x, y, LogarithmicInverseRetraction())
+end
+function inverse_retract(
```

Comment:
Why do we need these methods?

Reply:
It basically reverses the order:

The first (line 293) would allocate first and call `inverse_retract!` where the default retraction method is then resolved.
The second (here 296) would otherwise allocate and call `retract!` before then calling `log!`.
For both we don't want to go down the mutating route.

Reply:
Yes, I see.

Reply:
Thanks for checking/asking, it's always good to still check that the ideas are correct.